spirv_cross.cpp

```diff
@@ -5465,6 +5465,52 @@ void Compiler::analyze_interlocked_resource_usage()
 	}
 }
 
+// Helper function
+bool Compiler::check_internal_recursion(SPIRType &type, std::unordered_set<uint32_t> &checked_ids)
+{
+	// If this is not the base type (eg. it's a pointer or array), tunnel down
+	if (type.parent_type)
```

Comment:
It's more correct to look at type.self here. It will point to the non-pointer / non-array type.

Reply:
I took `if (type.parent_type)` from `CompilerMSL::mark_as_packable()`. Maybe it's not needed at all?

Removed it here.